internal/pkg/aws/acm/acm.go

```diff
@@ -66,23 +68,28 @@ func (a *ACM) ValidateCertAliases(aliases []string, certs []string) error {
 	}
 	for _, alias := range aliases {
 		if !validAliases[alias] {
-			return fmt.Errorf("%s is not a valid domain against %s", alias, strings.Join(certs, ","))
+			return &errInValidAliasAgainstCert{
+				certs:      certs,
+				alias:      alias,
+				sansOfCert: sansOfCert}
 		}
 	}
 	return nil
 }
 
-func (a *ACM) findValidAliasesAgainstCert(ctx context.Context, aliases []string, cert string) ([]string, error) {
+func (a *ACM) findValidAliasesAndAllowedDomainsAgainstCert(ctx context.Context, aliases []string, cert string) ([]string, []string, error) {
```

Comment:
Let's try to keep each method single purposed! What do you think of the following proposal:
```go
func (a *ACM) validDomainsOfCert(ctx context.Context, cert string) ([]string, err) {
   // Finds the domain and the SANs of the cert
}

func filterValidAliases(domains[]string, aliases[]string) )[]string {
  // Filter `aliases` based on `domains`
}

func (a *ACM) ValidateCertAliases(aliases []string, certs []string) error {
   // ...
   for i := range certs {
		cert := certs[i]
		g.Go(func() error {
			domains, err := a.validDomainsOfCert(ctx, cert)
			if err != nil {
				return err
			}
			validCertAliases := filterValidAliases()

                         mux.Lock()
			defer mux.Unlock()
			
			domainsOfCert[cert] = domains
			for _, alias := range validCertAliases {
				validAliases[alias] = true
			}
			return nil
                         
                         
	}
   // ...
}

```

Reply:
Done